Compare-NET-Objects/TypeComparers/IndexerComparer.cs

```diff
@@ -123,38 +123,25 @@ public void CompareIndexer(CompareParms parms, PropertyEntity info, PropertyEnti
             }
         }
 
-        private bool IndexersHaveDifferentLength(CompareParms parms, PropertyEntity info, PropertyEntity secondObjectInfo)
+        private bool IndexersHaveDifferentLength(CompareParms parms, PropertyEntity info, int indexerCount1, int indexerCount2)
         {
             if (info == null)
                 throw new ArgumentNullException("info");
 
-#if !NETSTANDARD
-            var type = info.ReflectedType;
-            var type2 = secondObjectInfo.ReflectedType;
-#else
-            var type = info.DeclaringType;
-            var type2 = secondObjectInfo.DeclaringType;
-#endif
-            if (type == null || type2 == null)
-                throw new ArgumentNullException("info");
-
-            int indexerCount1 = (int)type.GetProperty("Count").GetGetMethod().Invoke(parms.Object1, new object[] { });

```

Comment:
Do not know why we should use the reflection twice. We have alredy used reflection before to get this values.